File: OOP/OCP App/src/com/techlabs/ocp/solution/Circle.java
package com.techlabs.ocp.solution;

public class Circle implements AreaInterface {

	private double radius;

	public Circle(double radius) {
		super();
		this.radius = radius;
	}

	public double getRadius() {
		return radius;
	}

	@Override
	public double calculateArea() {
		double area = 2 * Math.PI * radius;
		return area;
	}

}

File: Node Js/AbstractClass/Circle.ts
import { Shape } from "./Shape";

export class Circle extends Shape{
    
    constructor(theX:number, theY:number, private _radius: number){
        super(theX, theY);
    }

    public getInfo():string{
        return super.getInfo() + " radius = " + this._radius;
    }

    calculateArea(): string {
        return "Area : " + Math.PI * this._radius * this._radius;
    }
}